runtime/hydradx/src/assets.rs

```diff
@@ -985,3 +992,141 @@ impl pallet_xyk::Config for Runtime {
 	type NonDustableWhitelistHandler = Duster;
 	type OracleSource = XYKOracleSourceIdentifier;
 }
+
+parameter_types! {
+	pub const ReferralsPalletId: PalletId = PalletId(*b"referral");
+	pub RegistrationFee: (AssetId,Balance, AccountId)= (NativeAssetId::get(), 222_000_000_000_000, TreasuryAccount::get());
+	pub const MaxCodeLength: u32 = 7;
+	pub const ReferralsOraclePeriod: OraclePeriod = OraclePeriod::Short;
+	pub const ReferralsSeedAmount: Balance = 100_000_000_000_000;
+}
+
+impl pallet_referrals::Config for Runtime {
+	type RuntimeEvent = RuntimeEvent;
+	type AuthorityOrigin = EnsureRoot<AccountId>;
+	type AssetId = AssetId;
+	type Currency = FungibleCurrencies<Runtime>;
+	type Convert = ConvertViaOmnipool;
+	#[cfg(not(feature = "runtime-benchmarks"))]
+	type PriceProvider =
+		OraclePriceProviderUsingRoute<Router, OraclePriceProvider<AssetId, EmaOracle, LRNA>, ReferralsOraclePeriod>;
+	#[cfg(feature = "runtime-benchmarks")]
+	type PriceProvider = ReferralsDummyPriceProvider;
+	type RewardAsset = NativeAssetId;
+	type PalletId = ReferralsPalletId;
+	type RegistrationFee = RegistrationFee;
+	type CodeLength = MaxCodeLength;
+	type TierVolume = ReferralsLevelTiers;
+	type SeedNativeAmount = ReferralsSeedAmount;
+	type WeightInfo = ();
+	#[cfg(feature = "runtime-benchmarks")]
+	type BenchmarkHelper = ReferralsBenchmarkHelper;
+}
+
+pub struct ConvertViaOmnipool;
+impl Convert<AccountId, AssetId, Balance> for ConvertViaOmnipool {
+	type Error = DispatchError;
+
+	fn convert(
+		who: AccountId,
+		asset_from: AssetId,
+		asset_to: AssetId,
+		amount: Balance,
+	) -> Result<Balance, Self::Error> {
+		let balance = Currencies::free_balance(asset_to, &who);
+		Omnipool::sell(RuntimeOrigin::signed(who.clone()), asset_from, asset_to, amount, 0)?;
```

Comment:
i would probably set some limit based on oracle just to be safe

Reply:
How could I determine such limit ?

Reply:
something like short price - 1% ?

Reply:
added